java-server-codegen fails compile with error: cannot find symbol ArrayNode.builder when adding cors and additionalAllowedHeaders in smithy model

Added additional headers - 
```
$version: "2"

namespace com.example

use aws.protocols#restJson1

/// Allows users to retrieve a menu, create a coffee order, and
/// and to view the status of their orders
@title("Coffee Shop Service")
@cors(
    origin: "*"
    additionalAllowedHeaders: [
        "*"
        "Access-Control-Allow-Origin"
        "Access-Control-Allow-Methods"
        "Access-Control-Allow-Headers"
        "Content-Type"
        "Authorization"
    ]
)
@restJson1
service CoffeeShop {
    version: "2024-08-23"
    operations: [
        GetMenu
    ]
    resources: [
        Order
    ]
}

/// Retrieve the menu
@http(method: "GET", uri: "/menu")
@readonly
operation GetMenu {
    output := {
        items: CoffeeItems
    }
}
```

Keeping everything else same as provided in quick start code( build, gradle files).

ran `./gradlew --refresh-dependencies clean build`

Fails with - 
```
server/build/smithyprojections/server/source/java-server-codegen/io/smithy/java/server/example/service/CoffeeShop.java:50: error: cannot find symbol
                  .withMember("additionalAllowedHeaders", ArrayNode.builder
                                                                   ^
    symbol:   variable builder
    location: class ArrayNode
  1 error
```

--- a/codegen/core/src/main/java/software/amazon/smithy/java/codegen/integrations/core/NodeWriter.java
+++ b/codegen/core/src/main/java/software/amazon/smithy/java/codegen/integrations/core/NodeWriter.java
@@ -127,7 +127,7 @@ record NodeWriter(JavaWriter writer, Node node) implements NodeVisitor<Void>, Ru
 
         @Override
         public Void arrayNode(ArrayNode arrayNode) {
-            writer.write("$T.builder", ArrayNode.class);
+            writer.write("$T.builder()", ArrayNode.class);
             writer.indent();
             for (var element : arrayNode.getElements()) {
                 writer.write(".withValue($C)", (Runnable) () -> element.accept(this));

